src/components/asthma/components/AsthmaActionPlanManager/AsthmaActionPlanManager.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useState } from 'react';
+import './AsthmaActionPlanManager.css';
+import { Button, LoadingIndicator, UnstyledButton } from '../../../presentational';
+import MyDataHelps from '@careevolution/mydatahelps-js';
+import { useInitializeView } from '../../../../helpers/Initialization';
+import sampleActionPlan from '../../assets/sample_aap.png'
+import { asthmaDataService } from '../../helpers';
+import { AsthmaActionPlan } from '../../model';
+import language from '../../../../helpers/language';
+
+export interface AsthmaActionPlanManagerProps {
+    previewState?: 'loading' | 'loaded without action plan' | 'loaded with action plan';
+    learnMoreUrl: string;
+    onViewActionPlan: (actionPlan: AsthmaActionPlan) => void;
+    editActionPlanSurveyName: string;
+    innerRef?: React.Ref<HTMLDivElement>;
+}
+
+export default function (props: AsthmaActionPlanManagerProps) {
+    const [loading, setLoading] = useState<boolean>(true);
+    const [actionPlan, setActionPlan] = useState<AsthmaActionPlan>();
+
+    const initialize = (retryCount: number = 0): void => {
+        setLoading(true);
+
+        if (props.previewState === 'loading') {
+            return;
+        }
+        if (props.previewState === 'loaded with action plan') {
+            setActionPlan({id: 'sample', url: sampleActionPlan});
+            setLoading(false);
+            return;
+        }
+        if (props.previewState === 'loaded without action plan') {
+            setActionPlan(undefined);
+            setLoading(false);
+            return;
+        }
+
+        asthmaDataService.loadParticipant().then(participant => {
+            if (participant.getActionPlanTaskRunUUID()) {
+                asthmaDataService.loadAsthmaActionPlan().then(actionPlan => {
+                    setActionPlan(actionPlan);
+                    if (actionPlan || retryCount >= 5) {
+                        setLoading(false);
+                    } else {
+                        setTimeout(() => initialize(retryCount++), 2000);
+                    }
+                });
+            } else {
+                setActionPlan(undefined);
+                setLoading(false);
+            }
+        });
+    };
+
+    useInitializeView(initialize, [], [props.previewState]);
+
+    const onLearnMore = (): void => {
+        if (props.previewState) return;
+        MyDataHelps.openApplication(props.learnMoreUrl, {modal: true});
```

Comment:
In terms of flexibility, does this have to be a URL? Idk what to expect in terms of this being repurposed, but it were to be reused, this could be difficult to workaround

Reply:
I think we typically implement drilldowns as separate modal views which launch via URL (as opposed to a within-view pop up).

Did you have something specific in mind for how a learn more drilldown might be implemented differently?

Reply:
Surveys can be easier for sites to create info pages vs urls.

Reply:
That is a good point.  As we discussed, I think I am going to leave this as a URL for now, since that is the only way it will be used for the foreseeable future, but I do think we should think about this on a more global level.

Perhaps we can think about bringing something like the `ActionDefinition` concept into MDH-UI from the view builder for cases where we want the user of the component to be able to define the action to be taken vs having the component define what the action taken will be.

Reply:
I think you would not need to bring in "ActionDefinition", you just need to expose function prop for handling the click and then in the view builder you can have that processed by the ActionDefinition.

(ActionDefinition feels inherently like a view builder thing; programmatically I can just handle the event)

Reply:
Fair enough.  That makes sense.

I think the bigger question will be a more philosophical one.  Should library components decide which actions to use by directly calling things like open application or start survey themselves, or should they all have event handler properties which are implemented externally to the component, thus allowing those events to be handled in a view-builder-configurable way using action definitions?

In other words, should we ever directly call open application or start survey from within in a component?  If yes, what is the criteria for deciding when the component should define which action to perform vs allowing it to be determined by an action definition config in the view builder?